### PR TITLE
[Docs] add colab tutorial

### PR DESCRIPTION
**Pull request by fangyixiao18**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

More convinient for users to use our codebase

## Modification

Add colab tutorial for basic usage
Part of content refers to https://github.com/open-mmlab/mmdetection/pull/7310 and MMCls

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by ZwwWayne** [comment]
image
imagenet.zip will make users consued whether this is the whole set of imagenet. We should use a more precise name, like imagenet_examples.zip

image
The style does not follow other OpenMMLab repos, which use high-level API to train the model, rather than using the script in colab. We should use high-level as many as we can, rather than simply run the scripts in colab.

File: demo/mmselfsup_colab_tutorial.ipynb [hunk 1]
```diff
@@ -0,0 +1,1589 @@
...
+      "id": "d76e94d0",
+      "metadata": {
+        "id": "d76e94d0"
+      },
+      "source": [
+        "# MMSelfSup Tutorial\n",
+        "In this tutorial, we will introduce the following content:\n",
+        "\n",
+        "- How to install MMSelfSup\n",
+        "- How to train the algorithm in MMSelfSup\n",
```
Comment (YuanLiuuuuuu) [inline demo/mmselfsup_colab_tutorial.ipynb]: `the algorithm` -> `algorithms`

File: demo/mmselfsup_colab_tutorial.ipynb [hunk 2]
```diff
@@ -0,0 +1,2426 @@
...
+        "## How to install MMSelfSup\n",
+        "\n",
+        "Before using MMSelfSup, we need to prepare the environment with the following steps:\n",
+        "\n",
+        "1. Install Python, CUDA, C/C++ compiler and git\n",
+        "2. Install PyTorch (CUDA version)\n",
+        "3. Install dependent codebase (mmcv, mmcls)\n",
+        "4. Clone mmselfsup source code from GitHub and install it\n",
+        "\n",
+        "Because this tutorial is on Google Colab, and the basic environment has been completed, we can skip the first two steps."
```
Comment (YuanLiuuuuuu) [inline demo/mmselfsup_colab_tutorial.ipynb]: changed to `all necessary packages have been installed`

File: demo/mmselfsup_colab_tutorial.ipynb [hunk 3]
```diff
@@ -0,0 +1,2426 @@
...
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "5b004a18",
+      "metadata": {
+        "id": "5b004a18"
+      },
+      "outputs": [],
+      "source": [
+        "# Install mmcv and mmcls\n",
+        "# !pip install mmcv -f https://download.openmmlab.com/mmcv/dist/cu111/torch1.10/index.html\n",
```
Comment (YuanLiuuuuuu) [inline demo/mmselfsup_colab_tutorial.ipynb]: remove `#` and the same below

Comment (fangyixiao18) [reply demo/mmselfsup_colab_tutorial.ipynb]: Actually it is recommended to use `mim` to install, which is above this line. It is just an optional command here.

**Review by YuanLiuuuuuu — APPROVED** [review]

**Pull request merged**